samples/authentication/app/src/main/java/de/telekom/authenticationdemo/utils/PKCEUtils.java

```diff
@@ -0,0 +1,45 @@
+package de.telekom.authenticationdemo.utils;
+
+import android.util.Base64;
+
+import net.openid.appauth.AuthorizationRequest;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Objects;
+
+import de.telekom.smartcredentials.core.authentication.configuration.PKCEConfiguration;
+
+/**
+ * Created by gabriel.blaj@endava.com at 10/6/2020
+ */
+public class PKCEUtils {
```

Comment:
I would like to avoid the word "utils". Please rename to something like `PkceProvider` or `PkceManager`. Also ... if `generatePKCEConfiguration()` is the only method used, make it static in order to avoid unnecessary initialization.

Reply:
changed it to PkceProvider